### PR TITLE
feat(snapshots): GetBackup details endpoint should support new dr

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Adds support to backup details endpoint for new DR.

#### Does this PR require a release note?

```release-note

```

### CODE REVIEW

File: pkg/handlers/backup.go [hunk 1]
```diff
@@ -130,22 +130,27 @@ func (h *Handler) ListInstanceBackups(w http.ResponseWriter, r *http.Request) {
 }
 
 type GetBackupResponse struct {
-	BackupDetail *snapshottypes.BackupDetail `json:"backupDetail"`
-	Success      bool                        `json:"success"`
-	Error        string                      `json:"error,omitempty"`
+	BackupDetails           []snapshottypes.BackupDetail `json:"backupDetails"`
+	Deprecated_BackupDetail *snapshottypes.BackupDetail  `json:"backupDetail"`
```
Comment (JGAntunes) [inline pkg/handlers/backup.go:134]: I wonder if this is better VS changing where we use this in the UI?
- https://github.com/replicatedhq/kots/blob/ea574d82fb8702318ef2c8c96831a474675237e0/web/src/components/RestoreSnapshotRow.jsx#L35
- https://github.com/replicatedhq/kots/blob/ea574d82fb8702318ef2c8c96831a474675237e0/web/src/components/snapshots/SnapshotDetails.jsx#L161

Else, should we have a store to clean this up once the UI is updated?

**Review by JGAntunes — APPROVED** [review]
LGTM, left a suggestion/question

File: pkg/handlers/backup.go (hunk 1 above)
Comment (emosbaugh) [reply pkg/handlers/backup.go:134]: Is something like this correct? 

```javascript
result.backupDetails && result.backupDetails.length > 0 ? result.backupDetails[0] : null
```

Comment (emosbaugh) [reply pkg/handlers/backup.go:134]: Or this?

```javascript
result.backupDetails?.[0]
```

Comment (emosbaugh) [reply pkg/handlers/backup.go]: updated

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**